app/src/main/java/net/squanchy/onboarding/location/LocationOnboardingActivity.java

```diff
@@ -85,69 +85,80 @@ private void disableUi() {
-                enableUi();
+                showNonFatalProximityError(Snackbar.make(contentRoot, R.string.proximity_error_permission_denied, Snackbar.LENGTH_LONG));
             }
 
             @Override
             public void locationProviderDenied() {
-                Timber.i("User denied location provider");
-                Snackbar.make(contentRoot, R.string.onboarding_error_location_denied, Snackbar.LENGTH_LONG).show();
-                enableUi();
+                showNonFatalProximityError(Snackbar.make(contentRoot, R.string.proximity_error_location_denied, Snackbar.LENGTH_LONG));
             }
 
             @Override
-            public void locationProviderFailed(LocationProviderPrecondition.FailureInfo failureInfo) {
-                Timber.i("Location provider check failed. Status: %s", failureInfo);
-                Snackbar.make(contentRoot, R.string.onboarding_error_location_failed, Snackbar.LENGTH_LONG)
-                        .setAction(R.string.onboarding_error_location_failed_action, view -> openLocationSettings())
-                        .show();
-                enableUi();
+            public void bluetoothDenied() {
+                showNonFatalProximityError(Snackbar.make(contentRoot, R.string.proximity_error_bluetooth_denied, Snackbar.LENGTH_LONG));
             }
 
             @Override
-            public void bluetoothDenied() {
-                Timber.i("User denied turning Bluetooth on");
-                Snackbar.make(contentRoot, R.string.onboarding_error_bluetooth_denied, Snackbar.LENGTH_LONG).show();
-                enableUi();
+            public void allChecksPassed() {
+                startRadarAndFinish();
             }
 
             @Override
             public void exceptionWhileSatisfying(Throwable throwable) {
                 Timber.e(throwable, "Exception occurred while checking");
-                Snackbar.make(contentRoot, R.string.onboarding_error_bluetooth_denied, Snackbar.LENGTH_LONG).show();
-                enableUi();
+                showNonFatalProximityError(Snackbar.make(contentRoot, R.string.proximity_error_bluetooth_denied, Snackbar.LENGTH_LONG));
+            }
+
+            @Override
+            public void recheckAfterActivityResult() {
+                optInToProximity();
             }
         };
     }
 
+    private void showFatalProximityError(@StringRes int snackbarText) {
+        Snackbar.make(contentRoot, snackbarText, Snackbar.LENGTH_LONG)
+                .addCallback(closeAfterFatalErrorIsDisplayed())
+                .show();
+    }
+
+    private Snackbar.BaseCallback<Snackbar> closeAfterFatalErrorIsDisplayed() {
+        return new Snackbar.Callback() {
+            @Override
+            public void onDismissed(Snackbar snackbar, int event) {
+                markPageAsSeenAndFinish();
+            }
+        };
+    }
+
+    private void showNonFatalProximityError(Snackbar snackbar) {
```

Comment:
Don't really care about that case that was covered by the snackbar actions, so I will simply uniform all the things.